src/internal/testutil/db.go

```diff
@@ -34,36 +37,47 @@ func OpenDBURL(t testing.TB, u pachsql.URL, password string) *pachsql.DB {
 
 // CreateEphemeralDB creates a new database using db with a lifetime scoped to the test t.
 func CreateEphemeralDB(t testing.TB, db *pachsql.DB, dbName string) {
-	_, err := db.Exec(`CREATE DATABASE ` + dbName)
+	ctx := pctx.TestContext(t)
+	err := CreateEphemeralDBNontest(ctx, db, dbName)
 	require.NoError(t, err)
 	if cleanup {
 		t.Cleanup(func() {
-			q := fmt.Sprintf("DROP DATABASE %s", dbName)
-			if db.DriverName() == "pgx" || db.DriverName() == "postgres" {
-				q += " WITH (FORCE)"
-			}
-			_, err := db.Exec(q)
+			err := CleanupEphemeralDB(ctx, db, dbName)
 			require.NoError(t, err)
 		})
 	}
 	t.Log("database", dbName, "successfully created")
 }
 
+// CreateEphermealDBNontest creates a new database outside of a test.
+func CreateEphemeralDBNontest(ctx context.Context, db *pachsql.DB, dbName string) error {
+	if _, err := db.ExecContext(ctx, `CREATE DATABASE `+dbName); err != nil {
+		return errors.Wrap(err, "create database")
+	}
+	return nil
+}
+
+// CleanupEphemeralDB cleans up a database.
+func CleanupEphemeralDB(ctx context.Context, db *pachsql.DB, dbName string) error {
+	q := fmt.Sprintf("DROP DATABASE %s", dbName)
+	if db.DriverName() == "pgx" || db.DriverName() == "postgres" {
+		q += " WITH (FORCE)"
+	}
+	if _, err := db.ExecContext(ctx, q); err != nil {
+		return errors.Wrap(err, "drop database")
+	}
+	return nil
+}
+
 // GenerateEphemeralDBName generates a random name suitable for use as a Postgres identifier
-func GenerateEphemeralDBName(t testing.TB) string {
+func GenerateEphemeralDBName() string {
```

Comment:
Hi Jon,
The PFS change looks good to me. Just want to discuss this: do you think is it meaningful to retain the original GenerateEphemeralDBName(t testing.TB) signature and add another helper function to do the rand.Reader.Read() logic, so that other functions call this helper function. Just like what you did to the CreateEphemeralDB. In this way, we can still test GenerateEphemeralDBName when running this file.
My first PR review ever and may not be helpful. Please let me know what you think. Thanks!

Reply:
Yeah, this can be cleaned up.  Bob and I were looking at this in more detail this morning and realized that most of the t.Cleanup calls can actually be defer db.Close(), and things like that.  So these can be unified into one, most likely.

Reply:
OK, I looked at this in more detail.   The thing Bob and I were talking about was whether or not we could `defer db.Close()` instead of passing the callback around to close things.  The answer is no, because we use the connection used to create the database to drop the database at the end.  (We could reconnect, but then if that fails, we can't clean up the database.  The current system has less ways to fail.)

For the GenerateEphemeralDBName call, we don't need a `*testing.T` object in there.  It was there to make the test fail if rand.Reader.Read failed, but `rand.Reader.Read` is pretty much failureproof.  On Linux it calls `getrandom (2)` which only errors when the buffer points to an invalid address or it's called with invalid arguments, which calling code cannot cause to happen.  For this reason, removing the `t` arg is fine (panic will fail the test too, but it will never panic), as it allows us to use this code from non-test code (which is what this PR needs).  I refactored all callers to not pass `t`.